### PR TITLE
User switching UX improvements

### PR DESCRIPTION
**Pull request by jakob223**
- [x] grey out the user switcher while a switch is in progress
- [x] show a progress indicator while the switch is in progress
- [x] don't show the login page in the middle of the switch
- [x] show "signed out" instead of greying out avatars
- [x] indicate logged-in users on the relogin page
- [x] on the relogin page, when you have an already-logged-in user selected, hide the password box.

Issue: Y2K-715

### CODE REVIEW

File: shared/login/relogin/index.d.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ export type Props = {
   onSignup: () => void
   onSomeoneElse: () => void
   error: string
+  hidePasswordBox: boolean
```
[the remark this answers is not in the document]
Comment (jakob223) [reply shared/login/relogin/index.d.ts]: then UMO it's ugly in the place where it's set - I thought this was the better balance. Will change though.

File: shared/login/relogin/index.desktop.tsx [hunk 2]
```diff
@@ -46,10 +48,10 @@ class Login extends React.Component<Props, State> {
   render() {
     const userRows = this.props.users
       .concat({hasStoredSecret: false, username: other})
-      .map(u => <UserRow user={u.username} key={u.username} />)
+      .map(u => <UserRow user={u.username} key={u.username} hasStoredSecret={u.hasStoredSecret} />)
 
     const selectedIdx = this.props.users.findIndex(u => u.username === this.props.selectedUser)
-
+    const needPassword = !this.props.hidePasswordBox
```
Comment (jakob223) [inline shared/login/relogin/index.desktop.tsx]: @jzila how's this compromise?

Comment (jzila) [reply shared/login/relogin/index.desktop.tsx]: But now you need to derive that variable twice.

Comment (jakob223) [reply shared/login/relogin/index.desktop.tsx]: hmm I think for some reason `needPassword` is much more acceptable to me than `showPasswordBox`. Renamed to that.

**Pull request merged**